### PR TITLE
Moved regenerateStamina, useStamina and exp multiplier (staminaChange as well) to Lua

### PR DESCRIPTION
**Pull request by ninjalulz**

### CODE REVIEW

File: data/events/scripts/player.lua [hunk 1]
```diff
@@ -110,6 +110,10 @@ local soulCondition = Condition(CONDITION_SOUL, CONDITIONID_DEFAULT)
 soulCondition:setTicks(4 * 60 * 1000)
 soulCondition:setParameter(CONDITION_PARAM_SOULGAIN, 1)
 
+local applyMultiplier = true
+local applyStaminaChange = true
+local nextUseStaminaTime = 0
```
Comment (diath) [inline data/events/scripts/player.lua]: Shouldn't this be unique per player?

Comment (ninjalulz) [inline data/events/scripts/player.lua]: It should, thanks for pointing it out. :P

**Comment by marksamman** [comment]
This changes the behavior of Player.addExperience, which makes it apply stamina and experience stage multiplier. I'm not sure what the best way to fix it is, perhaps checking source. You should also factor out the stamina code to its own local function in that file and decrease the indentation level on the code by immediatelly returning if configManager.getBoolean(configKeys.STAMINA_SYSTEM) is false.

**Comment by ninjalulz** [comment]
* Moved the stamina code to a local function,
* Player.addExperience is no longer affected by the stamina and experience stage multiplier.

**Comment by marksamman** [comment]
Thanks!

**Pull request merged**